Dev ble

Sabrina-V-Sousa opened the pull request:
ESP 32 code for starting the ble server and processing the message received from the RPi message.

Sabrina-V-Sousa committed:
Serial Communication: removed build files

Removed the generated files from the ESP 32 code build. The c file and 2
make files within ESP_UART_IN are all that is needed to build.

Sabrina-V-Sousa committed:
Serial Communication: Interrupt on Received Data

Modified the ESP c code to use interrupts. UART message is enqueued and
variables are modified based on message. Feedback loops through the
ouput sequenced determined by data until new variables indicate that it
should stop.

Code has been tested with a python script that loops through sample data
on the RPi. It cycles though a message for a distance in the further
range, then closer range then out of range. Output was as expected.

Sabrina-V-Sousa committed:
BLE

Created new branch for BLE work.

Sabrina-V-Sousa committed:
BLE: Connection and Basic Command of LED

Developped code to create the ble server on the esp in C code as well as
a basic python script that can find and connect to the esp device and
send a command to turn on an LED through GPIO 13.

This was tested by running the python code on a computer and was
successful. Since no connections are needed between the RPi and esp, it
will also work the same and it is better than the UART communication.

This tutorial was followed for creating a ble server:
https://innovationyourself.com/esp32-bluetooth-low-energy-tutorial/

Next steps are to have a KeepAlive message that maintains and checks the
ble connection or throw an error to alert the user that connection has
stopped (this is a safety concern). I will also change the message to
take the distance and threshold parameters. I also will send back an
acknowledgement.

Sabrina-V-Sousa committed:
BLE: Updates to ble server and test client script

BLE server has pattern feedback, it has a constant output when it is
disconnected and then different patterns (or OFF) when sensing heat. It
takes inputs for the signal bit, the 2 thresholds, and the calculated
distance. The esp 32 is the server and the RPi is the client. The BLE
sends keep alive messages so that their connection does not time out.

Testing was completed with the RPi client test script that connects to
the esp 32 server and loops through sending different sample inputs that
would trigger the different patterns.

Next steps are to implement this into the main code.

Sabrina-V-Sousa committed:
BLE: Changes made after integrating for last demo

Fixed logic according to the numbers that the RPi camera processing is
actually sending which was just accounting for negative numbers for
distance from hazard and positive numbers for within hazard. Added more
gpios for haptics, one haptic needs to be changed because it is not
available for reading and writing.

This has been tested and is functioning but disconnects sometimes. It is
likely an issue with how the RPi is connecting and not with this esp
code because the esp server works without disconnecting when connecting
to other devices like my computer, so it can be pushed.

Corbin-Jarrett commented:
Can you please add a newline at the end of all code files? You can see the symbol where it is missing in GitHub on the "files changed" tab

Corbin-Jarrett reviewed (changes requested):
See above comment for requested changes

sabrina-v-sousa commented:
sorry I meant to check that and then i forgot

Sabrina-V-Sousa committed:
BLE

Fixing whitespace and adding new lines in end of files.

No changes that require new testing.

Sabrina-V-Sousa committed:
BLE: Added newlines to ble and other files

Missed some newlines in last commit.

No required testing.

sabrina-v-sousa commented:
I fixed all the files so they all have newlines. I went through all BLE code files and fixed trailing whitespace too. Thanks!

Corbin-Jarrett commented:
Looks good to me, did you run it at least once to make sure nothing is broken?

sabrina-v-sousa commented:
Built it now and there were no issues!

Corbin-Jarrett reviewed (approved):
Did not review the code itself, but overall looks good.
